src/pages/Account/Tabs/ModulesTab/ViewCode.tsx

```diff
@@ -244,6 +270,26 @@ function ModuleHeader({
   );
 }
 
+function useStartingLineNumber(sourceCode?: string) {
+  const [searchParams] = useSearchParams();
+
+  if (!sourceCode) return 0;
+
+  const entryFunctionToHightlight = searchParams.get("entry_function");
+  if (entryFunctionToHightlight == null) return 0;
+
+  const lines = sourceCode.split("\n");
+  const lineNumber = lines.findIndex((line) =>
+    // TODO: improve line matching algorithm by using regex or something. Currently this will match falsely other entry functions that contain the entry function name in their name - e.g. `my_func_blue` will be matched when searching for `my_func`.
+    line.includes(`public entry fun ${entryFunctionToHightlight}`),
```

Comment:
thats what gpt told me `^public\sentry\sfun\s\{\$\{entryFunctionToHighlight\}\}$` lol

Reply:
hmm, that chatgpt output was pretty wrong in many ways, but I added something better :)